### PR TITLE
OCPBUGS-6727: Nutanix Hostname of the VM is not set when using DHCP network config

### PR DESCRIPTION
**Pull request by yanhua121**
Fix OCPBUGS-6727: Nutanix Hostname of the VM is not set when using DHCP network config

### CODE REVIEW

File: pkg/actuators/machine/machine_scope.go [hunk 1]
```diff
@@ -204,7 +208,46 @@ func (s *machineScope) getUserData() ([]byte, error) {
 		return nil, fmt.Errorf("The userData secret %s missing %s key", objKey, userDataSecretKey)
 	}
 
-	return userData, nil
+	ignData, err := s.getIgnitionData(userData)
+	if err != nil {
+		klog.Errorf("Failed to generate ignition user data. error: %w", err)
+		return nil, err
```
Comment (RadekManak) [inline pkg/actuators/machine/machine_scope.go:214]: Return the wrapped error. Do not only log it and return the parent error only.

File: go.mod [hunk 2]
```diff
@@ -21,6 +22,8 @@ require (
 	sigs.k8s.io/controller-tools v0.11.3
 )
 
+require github.com/vincent-petithory/dataurl v1.0.0
```
Comment (RadekManak) [inline go.mod:25]: Can this be with the other requires?

**Review by RadekManak — CHANGES_REQUESTED** [review]
Looks good, just some readability suggestions.

File: pkg/actuators/machine/machine_scope.go [hunk 3]
```diff
@@ -204,7 +208,46 @@ func (s *machineScope) getUserData() ([]byte, error) {
...
+	}
+	klog.V(3).Infof("%s: The ignition data: %s", s.machine.Name, string(ignData))
+
+	return ignData, nil
+}
+
+func (s *machineScope) getIgnitionData(userData []byte) ([]byte, error) {
```
[the remark this answers is not in the document]
Comment (yanhua121) [reply pkg/actuators/machine/machine_scope.go]: Moved.

File: pkg/actuators/machine/machine_scope.go (hunk 1 above)
Comment (yanhua121) [reply pkg/actuators/machine/machine_scope.go]: changed

File: go.mod (hunk 2 above)
Comment (yanhua121) [reply go.mod]: Moved.

**Comment by yanhua121** [comment]
@RadekManak Thanks for the review. Your comments are addressed.

**Review by elmiko — COMMENTED** [review]
/approve

**Pull request merged**